File: 49_group_anagrams/groupAnagrams.ts
function groupAnagrams(strs: string[]): string[][] {
  const hashMap = new Map();

  for (let s of strs) {
    const key = s.split("").sort().join("");

    if (hashMap.has(key)) {
      const value = hashMap.get(key);
      value.push(s);
    } else {
      hashMap.set(key, [s]);
    }
  }

  return Array.from(hashMap.values());
}

File: 49_group_anagrams/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}

        for s in strs:
            key = ''.join(sorted(s))
            val = hashMap.get(key, [])
            val.append(s)
            hashMap[key] = val
        
        return list(hashMap.values())
         